lib/src/genius.dart

```diff
@@ -331,27 +376,57 @@ class Genius {
             'Specified song does not contain lyrics. Rejecting.');
       }
 
-      songId = songInfo['id'];
+      return _getSongFromResponseInfo(
+          songInfo: songInfo, getFullInfo: getFullInfo);
+    } catch (e) {
+      return _verbosePrint('Error: ${e.toString()}');
+    }
+  }
 
-      if (songId != null && getFullInfo) {
-        Map<String, dynamic>? fullSongInfo = await song(songId: songId);
+  /// Searches for a song based on the provided [lyricsSnippet] and gets its lyrics returning [Song] in case it's successful and `null` otherwise .
+  ///
+  ///Args:
+  ///
+  /// `lyricsSnippet`: snippet of the lyrics of the song you are searching.
+  ///
+  /// `getFullInfo` (optional): Get full info for each song (slower).
+  ///
+  ///Example:
+  /// {@tool snippet}
+  ///
+  /// ```dart
+  ///Genius genius = Genius(accessToken: TOKEN);
+  /// Song? song = (await genius.searchSongByLyricsSnippet(
+  ///   lyricsSnippet: 'all the memories collected', getFullInfo: false));
+  ///if (song != null) {
+  ///  print(song.lyrics);
+  ///}
+  /// ```
+  /// {@end-tool}
+  Future<Song?> searchSongByLyricsSnippet(
+      {required String lyricsSnippet, bool getFullInfo = true}) async {
+    Map<String, dynamic>? response = await _httpClient.makeRequest(
+        url: searchByLyricsSnippetRoute,
+        query: {'q': lyricsSnippet},
+        headers: false);
+
+    if (response == null) {
+      return _verbosePrint('error making the request');
+    } else {
+      List<dynamic> sections = response['sections'];
 
-        if (fullSongInfo != null) {
-          songInfo = fullSongInfo;
-        } else {
-          _verbosePrint('error getting full song info');
-        }
-      }
+      Map<String, dynamic>? lyricSection =
+          firstWhereOrNull(sections, (element) => element['type'] == 'lyric');
 
-      String? url = (songInfo['url']);
+      if (lyricSection == null) {
+        return _verbosePrint('no song found');
+      } else {
+        Map<String, dynamic>? songInfo = firstWhereOrNull(lyricSection['hits'],
```

Comment:
maybe we should return all result and let the user decide wich one he want, because you could remember only a part of the song , and returning all results you will know all possibilyts

Reply:
That makes sense, will have to change the function name then. I will do it later